hashprobe/aeshash/aeshash_amd64.go

```diff
@@ -3,10 +3,25 @@
 package aeshash
 
 import (
-	"github.com/parquet-go/parquet-go/sparse"
+	"math/rand"
+	"unsafe"
+
 	"golang.org/x/sys/cpu"
+
+	"github.com/parquet-go/parquet-go/sparse"
 )
 
+const hashRandomBytes = unsafe.Sizeof(uintptr(0)) / 4 * 64
+
+var aeskeysched [hashRandomBytes]byte
+
+func init() {
+	key := (*[hashRandomBytes / 8]uint64)(unsafe.Pointer(&aeskeysched))
+	for i := range key {
+		key[i] = rand.Uint64()
```

Comment:
An issue with `rand` here is that it will be called before the program usually has an opportunity to set the random seed, which means the `aeskeysched` value will always be the same.

It looks like the Go runtime uses [chacha8](https://github.com/golang/go/blob/master/src/internal/chacha8rand/chacha8.go#L20) for these random values; maybe we can seed from `crypto/rand` by default, and assign the `aeskeysched` varaible to a known value in tests so we can have deterministic outputs?

Reply:
I'm not sure it's necessary to do anything else here because we're using the global random number generator (which is seeded by the runtime, not the program).

Reply:
> If Seed is not called, the generator is seeded randomly at program startup.

Prior to Go 1.20, the generator was seeded like Seed(1) at program startup. To force the old behavior, call Seed(1) at program startup. Alternately, set GODEBUG=randautoseed=0 in the environment before making any calls to functions in this package.

I didn't know this was changed in Go 1.20 👍